x-pack/plugins/ml/common/constants/trained_models.ts

```diff
@@ -28,6 +28,7 @@ export const SUPPORTED_PYTORCH_TASKS = {
   ZERO_SHOT_CLASSIFICATION: 'zero_shot_classification',
   TEXT_CLASSIFICATION: 'text_classification',
   TEXT_EMBEDDING: 'text_embedding',
+  TEXT_EXPANSION: 'text_expansion',
```

Comment:
I'm worried about the side effects of adding this model to the ML backend, without any specifics being implemented for `text_expansion` ([example](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/model_management/test_models/selected_model.tsx#L44)).

I'd rather handle the difference in ML trained models <=> Available trained models for Ent Search with some temporary code, e.g.:
```
  it('local LOCAL_SUPPORTED_PYTORCH_TASKS matches ml plugin', () => {
    // This test is expected to break when TEXT_EXPANSION is added to ml/common/constants/trained_models/SUPPORTED_PYTORCH_TASKS.
    // When that happens, please revert the code to this single line:
    // expect(SUPPORTED_PYTORCH_TASKS).toEqual(LOCAL_SUPPORTED_PYTORCH_TASKS);

    const TEMP_SUPPORTED_PYTORCH_TASKS = Object.fromEntries(Object.entries(LOCAL_SUPPORTED_PYTORCH_TASKS).filter((e) => e[0] !== 'TEXT_EXPANSION'));
    expect(SUPPORTED_PYTORCH_TASKS).toEqual(TEMP_SUPPORTED_PYTORCH_TASKS);
  });
```

What do you think?
```

Reply:
I'm fine with that. I would also add a comment to our [SUPPORTED_PYTORCH_TASKS](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts#L29) with `TEXT_EXPANSION` at the bottom and callout that we explicity added it even though its not in the ml plugin  to support ELSER.

But I'd also want to double check with the ml team on if he should be added to the ml plugin's `SUPPORTED_PYTORCH_TASKS`

Reply:
Ok, added comments both in `index.ts` and `index.test.ts`.

Reply:
I rebased the branch after the ML team modularized `SUPPORTED_PYTORCH_TASKS` and related stuff in [this PR](https://github.com/elastic/kibana/pull/154046). With that the local copy is no longer needed so it was removed.